src/ratbag_emu/protocol/util/profile.py

```diff
@@ -1,47 +1,89 @@
 from collections import namedtuple
 
+
 '''
 Imeplemnts a profile with all the mouse settings
 '''
 class Profile(object):
     Led = namedtuple('Led', ['red', 'green', 'blue'])
 
-    def __init__(self, obj=None):
+    class Button(object):
+        def __init__(action_type, val=None):
+            self.action_type = action_type
+            self.button = None
+            self.special = None
+            self.macro = None
+
+            setattr(self, self.action_type, val)
```

Comment:
usually, when these kind of things are being written, it's because you are not using the classes correctly:

You should define 3 `ButtonAction` classes, one for each type. Then each class would know how to treat its `value`.

Reply:
Can I make it an if? 3 separate classes seems like overkill to me.

Reply:
classes do not cost anything in python.

The `if` would be clearer for the reviewer, but not beneficial as the `setattr` is more efficient.

I still maintain we should use subclassing for that particular case. If you really do not want, please add a `TODO` comment and move on

Reply:
Okay, that is fine by me. I still don't understand your reasoning. If and when you have a bit of time to explain please do. If I am not wrong these fields don't need any differences in the processing of the data. Actually, I think it would be best to just convert this into a `namedtuple` like `Led`.